models/app.model.js

```diff
@@ -4,4 +4,18 @@ exports.readTopics = () => {
     return db.query(`SELECT * FROM topics;`).then(({ rows }) => {
       return rows;
     })
-  }
\ No newline at end of file
+  }
+
+  exports.fetchArticleById = (article_id) =>{
+    if(isNaN(parseInt(article_id))){
+      return Promise.reject({ status: 400, msg: "Bad Request" });
+    }
+    return db
+    .query(`SELECT * FROM articles WHERE article_id=$1`, [article_id])
+    .then(({ rows }) => {
+    if(rows.length === 0){
+      return Promise.reject({code: "INVALID_ID", msg: "Article Doesn't Exist"})
+    }
```

Comment:
You should only end up here if the rows are empty. If the rows are empty then it will be because you have entered a valid id but it's not found anything. This wouldn't be an invalid_id - it would be not found. Instead of doing code I would send status: 404